rename_imports.py

```diff
@@ -0,0 +1,69 @@
+from LSP.plugin.core.file_watcher import FileWatcher
+from LSP.plugin.core.file_watcher import FileWatcherEvent
+from LSP.plugin.core.file_watcher import FileWatcherProtocol
+from LSP.plugin.core.typing import List, Type
+import os
+import sublime
+
+
+class HandleRenameImport(FileWatcherProtocol):
+    def __init__(self, window: sublime.Window, root_path: str, file_watcher: Type[FileWatcher]):
+        self.window = window
+        self.events = []  # type: List[FileWatcherEvent]
+        self.file_watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=self
+        )
+
+    def clean_up(self):
+        self.file_watcher.destroy()
+
+    def on_file_event_async(self, events: List[FileWatcherEvent]) -> None:
+        self.events.extend(events)
+        sublime.set_timeout_async(self._check_if_there_was_a_rename, 10)
+
+    def _check_if_there_was_a_rename(self):
+        if len(self.events) == 2:
+            create_event = next((event for event in self.events if event[0] == 'create'), None)
+            delete_event = next((event for event in self.events if event[0] == 'delete'), None)
```

Comment:
This seems very fragile and implementation dependent. The logic should just be: if there are two sibling events where one is 'create' and one is 'delete' then trigger the rename.

Same for the timeout that calls this method. There is no guarantee that both events will come within 10ms and if they don't, we'll clear the list and miss the whole rename.

Reply:
And to clarify, I meant without checking if there are exactly 2 events as that is not guaranteed.

Reply:
> The logic should just be: if there are two sibling events where one is 'create' and one is 'delete' then trigger the rename.

We cannot rely on sibling events. :)

Here are the logs when renaming multiple files at once on Linux,  .
the folder structure is the following:
```
├── test
│   ├── file1.ts
│   ├── file2.ts
│   └── file3.ts
```

if we rename multiple files at once,
we will have the following events logged:
```
events [('delete', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file1.ts')]
events [('delete', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file2.ts')]
events [('delete', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file3.ts')]
events [('create', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file11.ts')]
events [('create', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file21.ts')]
events [('create', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file31.ts')]
```
<details>
<summary>
see the gif
</summary>

![output](https://user-images.githubusercontent.com/22029477/148395033-c390f834-2b93-42c6-81bd-3c8fc89aabe4.gif)

</details>

So to jump to my point is, 
we cannot rely on two sibilling events, 
as that would result in incorrect behavior. 

In the above example, it would mean to rename `file3.ts` to `file11.ts`, which is not correct.
```
events [('delete', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file3.ts')]
events [('create', '/home/predragnikolic/.config/sublime-text/Packages/LSP-typescript/test/file11.ts')]
```

Reply:
In that case it seems like we need more sophisticated solution. Something similar to what is mentioned in the link you've posted before (https://github.com/paulmillr/chokidar/issues/303#issuecomment-903710306)

Reply:
I also can imagine that 3 renames in a row would not be handled well by the quick panel that we are triggering.

Reply:
I intentionally didn't want to support multiple file renames at once. only because I do not want to spam the user will prompts.

I only wanted to cover the case when a user renames a single file.

Reply:
> In that case it seems like we need more sophisticated solution. Something similar to what is mentioned in the link you've posted before ...

Will improve the logic by checkin the `iNode` number of a file. :)

Reply:
well,
I tried to use `os.stat(file_name).st_ino` to get the ino number for a file.

But I quickly run into problems.

Here was my soloution:

![image](https://user-images.githubusercontent.com/22029477/148829524-3a21bfcc-82cd-4b1d-aff1-58e94d7a2135.png)

```
events_with_ino = [(event[0], event[1], os.stat(event[1]).st_ino) for event in events]
#					^^^^^^^^ event_type
#							  ^^^^^^^^ file_name
#										^^^^^^^^^^^^^^^^^^^^^^^^ ino number for that file_name
```
I tried to attach an ino number for a file_name,
but this works only for files that exist.

That said os.stat(event[1]).st_ino will work only for `create` or `change` events,
but not for when the event type is `delete`.
Because by the time we receive the delete event in python,
the file no longer exists,
so calling `os.stat(file_name_that_no_longer_exist).st_ino` will throw.  

----

Ideally I would shift this logic to be the responsibility of the file_watcher, 
so we have an interface like this:
https://github.com/neoclide/coc-tsserver/blob/master/src/server/features/updatePathOnRename.ts#L33

Reply:
Supporting this in the watcher is the way to go but not sure if we would be able to do it in `LSP-file-watcher-chokidar`. Might need to be implemented upstream in `chokidar` instead.